[JIT] Remove dead store in unpickler.cpp

SplitInfinity opened the pull request:
Test Plan: Continuous integration.

Differential Revision: D22259289

suo reviewed (changes requested):
looks like MSVC wants that variable?

SplitInfinity commented:
No, I got too trigger happy - the dead store is `idx++` not `requires_grad` :(

Commit:
[JIT] Remove dead store in unpickler.cpp (#40625)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/40625

Test Plan: Continuous integration.

Differential Revision: D22259289

fbshipit-source-id: 7304688af8b9aa277c7ce672118cfa0f655ec11d
